app/histories_v2.go

```diff
@@ -39,6 +39,14 @@ func HistoriesV2Handler(app *App) func(c echo.Context) error {
 			topicMessages := mongoclient.GetMessagesV2(c, topic, from, limit, collection)
 			messages = append(messages, topicMessages...)
 		}
+
+		gameId := messages[0].GameId
+		metricTagMap := c.Get("metricTagsMap").(map[string]interface{})
+		if metricTagMap != nil {
+			metricTagMap["gameID"] = gameId
+			logger.Logger.Debugf("provided gameID: %s", metricTagMap["gameID"])
+		}
```

Comment:
Here you should use the safe type assertion, i. e.:

```suggestion
		if metricTagsMap, ok := c.Get("metricTagsMap").(map[string]interface{}); ok {
				metricTagsMap["gameID"] = gameId
		}
```

That's because if "metricTagsMap" value was not present, it would return `nil`, which throws an error when a type assertion is issued upon.

Also checking for `nil` is not necessary.

Moreover, remove the `debug` log, I think it would be a leftover from your tests, right?